fendermint/eth/api/src/cache.rs

```diff
@@ -3,27 +3,73 @@
 
 use std::sync::{Arc, Mutex};
 
-use crate::state::ActorType;
 use anyhow::Context;
 use cid::Cid;
-use fendermint_rpc::client::FendermintClient;
-use fendermint_rpc::query::QueryClient;
-use fendermint_vm_message::query::FvmQueryHeight;
+use lru_time_cache::LruCache;
+use tendermint_rpc::Client;
+
 use fvm_shared::{
     address::{Address, Payload},
     ActorID,
 };
-use lru_time_cache::LruCache;
-use tendermint_rpc::Client;
+
+use fendermint_rpc::client::FendermintClient;
+use fendermint_rpc::query::QueryClient;
+use fendermint_vm_message::query::FvmQueryHeight;
+
+use crate::state::ActorType;
+
+// The `LruCache` is wrapped in `Mutex` beause even reading requiers mutation.
+#[derive(Clone)]
+pub struct Cache<K, V> {
+    cache: Arc<Mutex<LruCache<K, V>>>,
+}
+
+impl<K, V> Cache<K, V>
+where
+    K: Ord + Clone,
+    V: Clone,
+{
+    pub fn new(capacity: usize) -> Self {
+        Self {
+            cache: Arc::new(Mutex::new(LruCache::with_capacity(capacity))),
```

Comment:
Does it come with some default time to live as well?

Reply:
No, it's optional.